### ISSUE TITLE
crash when I open "Running Processes"

### ISSUE THREAD

**Issue by weisuofayubc** [MuntashirAkon/AppManager#1325]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

```
03-27 11:28:40.376  1857  1899 I FlexibleTaskController: releaseFlexibleTaskCaptionView task=Task{5f92cc #118 type=standard A=10276:io.github.muntashirakon.AppManager}
03-27 11:28:40.376  1857  1899 I FlexibleTaskController: java.lang.Throwable
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.FlexibleTaskController.releaseFlexibleCaptionView(FlexibleTaskController.java:1224)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.FlexibleWindowManagerService.onFlexibleWindowTaskVanished(FlexibleWindowManagerService.java:777)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerControllerExtImpl.onTaskVanished(TaskOrganizerControllerExtImpl.java:105)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerController$TaskOrganizerCallbacks.onTaskVanished(TaskOrganizerController.java:158)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerController$TaskOrganizerPendingEventsQueue.dispatchPendingEvent(TaskOrganizerController.java:316)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerController$TaskOrganizerPendingEventsQueue.dispatchPendingEvents(TaskOrganizerController.java:302)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerController.dispatchPendingEvents(TaskOrganizerController.java:1050)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:212)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:107)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator.$r8$lambda$aHNu1uhcqxihX5NZc4McDDQPAyw(WindowAnimator.java:0)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1550)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1561)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.view.Choreographer.doCallbacks(Choreographer.java:1117)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.view.Choreographer.doFrame(Choreographer.java:992)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1535)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.os.Handler.handleCallback(Handler.java:958)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.os.Handler.dispatchMessage(Handler.java:99)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.os.Looper.loopOnce(Looper.java:257)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.os.Looper.loop(Looper.java:368)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at android.os.HandlerThread.run(HandlerThread.java:67)
03-27 11:28:40.376  1857  1899 I FlexibleTaskController:        at com.android.server.ServiceThread.run(ServiceThread.java:46)
03-27 11:28:40.376  2464  2633 D ShellTaskOrganizerExt: visibleFlexibleTaskInfos delete:118
03-27 11:28:40.376  1857  1899 D OplusFlexibleWindowMinimizedManagerHelper: onTaskVanished taskInfo:TaskInfo{userId=0 taskId=118 displayId=-1 isRunning=false baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=io.github.muntashirakon.AppManager/.main.SplashActivity mCallingUid=10184 } baseActivity=null topActivity=null origActivity=null realActivity=ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.main.SplashActivity} numActivities=0 lastActiveTime=2889082 supportsSplitScreenMultiWindow=false supportsMultiWindow=false supportsPocketStudioMultiWindow=true isInFlexibleEmbedded = false uid = -1 pid = -1 resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{RemoteToken{c7d79e7 Task{5f92cc #118 type=standard A=10276:io.github.muntashirakon.AppManager}}} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=null topActivityInfo=null launchCookies=[] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false topActivityInSizeCompat=false topActivityInOplusCompatMode=false topActivityEligibleForLetterboxEducation= false topActivityLetterboxed= false isFromDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=-1 topActivityLetterboxHeight=-1 locusId=null displayAreaFeatureId=-1 cameraCompatControlState=hidden locusId= null topActivityLetterboxInsets=Rect(0, 0 - 0, 0)} task:Task{5f92cc #118 type=standard A=10276:io.github.muntashirakon.AppManager}
03-27 11:28:40.377  1857  1899 I FlexibleTaskController: releaseFlexibleTaskCaptionView task=Task{5f92cc #118 type=standard A=10276:io.github.muntashirakon.AppManager}
03-27 11:28:40.377  1857  1899 I FlexibleTaskController: java.lang.Throwable
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.FlexibleTaskController.releaseFlexibleCaptionView(FlexibleTaskController.java:1224)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.FlexibleWindowManagerService.onFlexibleWindowTaskVanished(FlexibleWindowManagerService.java:777)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerControllerExtImpl.onTaskVanished(TaskOrganizerControllerExtImpl.java:105)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerController$TaskOrganizerPendingEventsQueue.dispatchPendingEvent(TaskOrganizerController.java:319)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerController$TaskOrganizerPendingEventsQueue.dispatchPendingEvents(TaskOrganizerController.java:302)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.TaskOrganizerController.dispatchPendingEvents(TaskOrganizerController.java:1050)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:212)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:107)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator.$r8$lambda$aHNu1uhcqxihX5NZc4McDDQPAyw(WindowAnimator.java:0)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1550)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1561)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.view.Choreographer.doCallbacks(Choreographer.java:1117)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.view.Choreographer.doFrame(Choreographer.java:992)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1535)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.os.Handler.handleCallback(Handler.java:958)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.os.Handler.dispatchMessage(Handler.java:99)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.os.Looper.loopOnce(Looper.java:257)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.os.Looper.loop(Looper.java:368)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at android.os.HandlerThread.run(HandlerThread.java:67)
03-27 11:28:40.377  1857  1899 I FlexibleTaskController:        at com.android.server.ServiceThread.run(ServiceThread.java:46)

```

### Device info

- Device: OnePlus ACE2V
- OS Version: colorOS 14
- App Manager Version: 4.0.0
- Mode:  Root

**Comment by MuntashirAkon** [comment]
App Manager is yet to release 4.0.0. Do you mean a specific release like alpha01, alpha02, debug, etc.?

**Comment by weisuofayubc** [comment]
yes. I download master code and build it myself. By the way, I try with 3.1.5 and it works

**Comment by MuntashirAkon** [comment]
I see. But you haven't shared the actual crashlog. Everything you shared here are only information. They do not correspond to the actual crash.

**Comment by weisuofayubc** [comment]
Oh yes, my bad. Now I run with debug version and it has more clear log..
```
2024-03-27 21:31:09.195 15881-15881 AndroidRuntime          ioo...untashirakon.AppManager.debug  D  Shutting down VM
2024-03-27 21:31:09.196 15881-15881 AndroidRuntime          ioo...untashirakon.AppManager.debug  E  FATAL EXCEPTION: main
                                                                                                    Process: ioo.github.muntashirakon.AppManager.debug, PID: 15881
                                                                                                    java.lang.ArithmeticException: divide by zero
                                                                                                    	at io.github.muntashirakon.AppManager.runningapps.RunningAppsAdapter.lambda$onBindViewHolder$1(RunningAppsAdapter.java:153)
                                                                                                    	at io.github.muntashirakon.AppManager.runningapps.RunningAppsAdapter$$ExternalSyntheticLambda3.run(Unknown Source:6)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:257)
                                                                                                    	at android.os.Looper.loop(Looper.java:368)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8826)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
2024-03-27 21:31:09.198 15881-15881 Compatibil...geReporter ioo...untashirakon.AppManager.debug  D  Compat change id reported: 160794467; UID 10321; state: ENABLED
2024-03-27 21:31:09.198 15881-15881 Compatibil...geReporter ioo...untashirakon.AppManager.debug  D  Compat change id reported: 236704164; UID 10321; state: ENABLED
2024-03-27 21:31:09.200 15881-15881 AndroidRuntime          ioo...untashirakon.AppManager.debug  E  Error reporting crash
                                                                                                    java.lang.IllegalArgumentException: ioo.github.muntashirakon.AppManager.debug: Targeting U+ (version 34 and above) disallows creating or retrieving a PendingIntent with FLAG_MUTABLE, an implicit Intent within and without FLAG_NO_CREATE and FLAG_ALLOW_UNSAFE_IMPLICIT_INTENT for security reasons. To retrieve an already existing PendingIntent, use FLAG_NO_CREATE, however, to create a new PendingIntent with an implicit Intent use FLAG_IMMUTABLE.
                                                                                                    	at android.os.Parcel.createExceptionOrNull(Parcel.java:3091)
                                                                                                    	at android.os.Parcel.createException(Parcel.java:3071)
                                                                                                    	at android.os.Parcel.readException(Parcel.java:3054)
                                                                                                    	at android.os.Parcel.readException(Parcel.java:2996)
                                                                                                    	at android.app.IActivityManager$Stub$Proxy.getIntentSenderWithFeature(IActivityManager.java:6568)
                                                                                                    	at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:549)
                                                                                                    	at android.app.PendingIntent.getActivity(PendingIntent.java:530)
                                                                                                    	at android.app.PendingIntent.getActivity(PendingIntent.java:494)
                                                                                                    	at androidx.core.app.PendingIntentCompat.getActivity(PendingIntentCompat.java:114)
                                                                                                    	at io.github.muntashirakon.AppManager.misc.AMExceptionHandler.uncaughtException(AMExceptionHandler.java:56)
                                                                                                    	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1071)
                                                                                                    	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1066)
                                                                                                    	at com.android.internal.os.RuntimeInitExtImpl.uncaughtExceptionExt(RuntimeInitExtImpl.java:68)
                                                                                                    	at com.android.internal.os.RuntimeInit$LoggingHandler.uncaughtException(RuntimeInit.java:128)
                                                                                                    	at java.lang.Thread.dispatchUncaughtException(Thread.java:2300)
                                                                                                    Caused by: android.os.RemoteException: Remote stack trace:
                                                                                                    	at com.android.server.am.ActivityManagerService.getIntentSenderWithFeatureAsApp(ActivityManagerService.java:5908)
                                                                                                    	at com.android.server.am.ActivityManagerService.getIntentSenderWithFeature(ActivityManagerService.java:5851)
                                                                                                    	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3251)
                                                                                                    	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:3005)
                                                                                                    	at android.os.Binder.execTransactInternal(Binder.java:1392)
2024-03-27 21:31:09.211  1857-2329  UAH_JNI                 system_server                        E  uah_event_acquire ! cmdId: 0 ,strPkg: ioo.github.muntashirakon.AppManager.debug, strIdent: com.android.server.wm, timeout: 600
2024-03-27 21:31:09.213 15881-17172 JavaBinder              ioo...untashirakon.AppManager.debug  W  ibinderForJavaObject: 0x74031e0b30 is not a Binder object
2024-03-27 21:31:09.213 15881-17172 Parcel                  ioo...untashirakon.AppManager.debug  E  Native binder in markForBinder is null for non-null jobject
```

**Comment by weisuofayubc** [comment]
I disable swap on my phone since I have 16GB memory..

**Comment by MuntashirAkon** [comment]
Fixed in de7b3bfd183af0f3d42f19cf630442e35f08d2c6
